### ISSUE TITLE
Top-level enum definition (e.g. OuterEnum) is not processed properly in html doc and python client code generation

### ISSUE THREAD

**Issue by easz** [swagger-api/swagger-codegen#5375]
**test input:** 
   [test_enum.yaml](https://gist.github.com/easz/56ceba60bd06276c9891544e8b93dc7d)
   parameter is defined with a referenced type which references another string enum type.

**test with swagger-codegen 2.2.2:**
   - $ java -jar swagger-codegen-cli-2.2.2.jar generate -i [test_enum.yaml](https://gist.github.com/easz/56ceba60bd06276c9891544e8b93dc7d) -l python -o python-client
    and
   - $ java -jar swagger-codegen-cli-2.2.2.jar generate -i [test_enum.yaml](https://gist.github.com/easz/56ceba60bd06276c9891544e8b93dc7d) -l html -o doc-html

**test result:**
  - Without FOO, BAR enum information as expected
    - python-client/swagger_client/models/[an_enum_type.py](https://gist.github.com/easz/5dc0eb12d5113a29b388f1448cf3d72d)
    - doc-html/[index.html](https://gist.github.com/easz/e1a9d793c32e73532ea52452fa0fd3d0)

**note:**
   - java, javascript, html2 generation are OK

**Comment by wing328** [comment]
@easz Python API client lacks support for enum at the moment. May I know if you've time to contribute the enhancement? I can show you some good starting points as we've done it in other languages: PHP, C#, Java, JS, etc

**Comment by easz** [comment]
Sure,

**Comment by easz** [comment]
@wing328  do you know what kind of template variables (e.g. isEnum, allowableValues ..etc) when codegen parses **outerEnum** **property** inside the **model** **Enum_Test**.
Apparently there is **!isEnum and !hasValidation**

It would be easier to fix it on the model property level as in top-level enum itself.

```
 definitions:
    Enum_Test:
      type: object
      properties:
        outerEnum:
          $ref: '#/definitions/OuterEnum'
    OuterEnum:
      type: "string"
      enum:
        - "placed"
        - "approved"
```

**Comment by easz** [comment]
The actual model information of a referenced model is not used for template generation.

Namely the **RefModel** is at the moment not processed in **DefaultCodegen**.**fromModel** and will also not be processed properly during **DefaultGenerator.generateModels** 

The intuitive solution would be just to assign the actual Model for RefModel where the TODO is...
@wing328 does it fit to swagger design?


```
    public CodegenModel fromModel(String name, Model model, Map<String, Model> allDefinitions) {
        if (model instanceof ArrayModel) {
             ...
        } else if (model instanceof RefModel) {
            // TODO
        }
```

**Comment by easz** [comment]
The easiest solution, similar to some other languages, is just to define the allowable enum values in the model (e.g. OuterEnum). For this I can provide a PR.

For example
```
class OuterEnum(object):
    PLACED = "placed"
    APPROVED = "approved"
    DELIVERED = "delivered"
```
whether it is suitable or friendly for python programming I am not sure. I am python beginner.

Another solution: to have explicit python Enum class
however the build-in Enum type is not supported in Python 2.
Probably let such enum model extend built-in python str class and limit the allowable value

Another solution: assign such enum properly explicitly in the properties instead of a ref.
so that for example EnumTest has direct OuterEnum in its value/property list instead of just a RefModel.

**Comment by wing328** [comment]
Closed via #5403, #5512
